src/foam/nanos/auth/CachedAuthServiceProxy.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.auth',
+  name: 'CachedAuthServiceProxy',
+  extends: 'foam.nanos.auth.ProxyAuthService',
+  documentation: `
+    An auth service decorator that caches permission checks. The cache gets
+    invalidated whenever the imported user's group changes.
+  `,
+  imports: [
+    'user',
+  ],
+  properties: [
+    {
+      class: 'Map',
+      name: 'cache',
+    },
+  ],
+  methods: [
+    function init() {
+      this.onDetach(this.user$.dot('group').sub(function() {
```

Comment:
User never changes, so you could just do this.user.group$.sub()

Reply:
Isn't this technically a little more robust if user ever does change in the future? What's the downside of using .dot? I wouldn't think the overhead would be that much more but I could be wrong.

Reply:
Yes, it is more robust, but all of our code is designed assuming user doesn't change, so you could go with the simpler code.